### PR TITLE
[FPGA] Add support for the FPGA simulator

### PR DESCRIPTION
**Pull request by intel-jisheng1**
Add support for the FPGA simulator to run the code samples for max_interleaving, loop_ivdep, loop_unroll

Uses ext::intel::fpga_simulator_selector class to choose the simulator.

# Existing Sample Changes
DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/
DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/
DirectProgramming/DPC++FPGA/Tutorials/Features/loop_unroll/

## Description

Add support for running these code samples on the FPGA simulator. Reduced data sizes to ensure reasonable execution times, as the simulator is 1000-10000x slower than the hardware

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

This has been run manually on Linux. Windows testing is on going.

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [X] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/src/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +23,9 @@ endif()
 #    For this reason, FPGA backend flags must be passed as link flags in CMake.
 set(EMULATOR_COMPILE_FLAGS "-Wall -fintelfpga -DFPGA_EMULATOR")
 set(EMULATOR_LINK_FLAGS "-fintelfpga")
+set(SIMULATOR_COMPILE_FLAGS "-Wall -fintelfpga -Xssimulation -DFPGA_SIMULATOR ${USER_SIMULATOR_FLAGS}")
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/src/CMakeLists.txt:26]: Please pull the latest version of the code samples, you'll see the compile and link flags have changed slightly to add `-fsycl`. So you'll have to add this to your flags as well (in this CMakeFile and the others)

Comment (yuguen-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/src/CMakeLists.txt:26]: Ajay thinks it is better to use USER_HARDWARE_FLAGS for both simulation and HW, so please make this change as well

**Review by yuguen-intel — CHANGES_REQUESTED** [review]
Looks good to me, please just fetch upstream and apply the associated changes to your simulation cmake targets (basically adding `-fsycl`) + the USER_SIMULATOR_FLAGS comment
Thanks!

File: DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/src/CMakeLists.txt (hunk 1 above)
Comment (intel-jisheng1) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/loop_ivdep/src/CMakeLists.txt]: updated.

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**